misaligned.c

```diff
@@ -1,21 +1,55 @@
 #include <stdio.h>
 #include <assert.h>
+#include <string.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+char line[1000];
+char reference_line[1000];
+
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+ 
+int printColorMap(char *line)
+{
+    int pairno =0;
     int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+    for(i = 0; i < 5; i++)
+    {
+        for(j = 0; j < 5; j++)
+        {
+            pairno += sprintf(&line[pairno],"%d | %s | %s\n", ((i * 5 + j)+1), (majorColor[i]), (minorColor[j]));           
         }
     }
     return i * j;
 }
 
-int main() {
-    int result = printColorMap();
+void colorcode_ReferenceManual_Display(char *reference_line)

```

Comment:
Two very similar functions?

Reply:
one is the original func with buggy code and one more is the reference func and then i am performing strcmp @sudeeprp